sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1596,7 +1596,9 @@ object ReplaceDeduplicateWithAggregate extends Rule[LogicalPlan] {
       // we append a literal when the grouping key list is empty so that the result aggregate
       // operator is properly treated as a grouping aggregation.
       val nonemptyKeys = if (keys.isEmpty) Literal(1) :: Nil else keys
-      Aggregate(nonemptyKeys, aggCols, child)
+      val newAgg = Aggregate(nonemptyKeys, aggCols, child)
+      val attrMapping = d.output.zip(newAgg.output)
```

Comment:
`d.output` is actually `child.output`, but `newAgg.output` may have different number of columns if `keys.isEmpty`. how about
```
if (keys.isEmpty) {
  Aggregate(Seq(Literal(1)), aggCols, child)
} else {
  val newAgg  = ...
  ...
}
```

Reply:
Looks the length of `Aggregate.output` is `aggCols.length` and `aggCols.length` is the same with `Deduplicate.child.length`?

Reply:
ah sorry, it's the grouping key, nvm.